pkg/domain/config.go

```diff
@@ -0,0 +1,40 @@
+package domain
+
+import (
+	"context"
+)
+
+// PostgresConfig contains the Postgres database configuration arguments
+type PostgresConfig struct {
+	Hostname     string `json:"hostname"`
+	Port         string `json:"port"`
+	Username     string `json:"username"`
+	Password     string `json:"password"`
+	DatabaseName string `json:"databasename"`
+}
+
+// Name is used by the settings library to replace the default naming convention.
+func (c *PostgresConfig) Name() string {
+	return "Postgres"
+}
+
+// PostgresConfigComponent satisfies the settings library Component API,
+// and may be used by the settings.NewComponent function.
+type PostgresConfigComponent struct{}
+
+// Settings populates a set of defaults if none are provided via config.
+func (*PostgresConfigComponent) Settings() *PostgresConfig {
+	return &PostgresConfig{}
+}
+
+// New constructs a PostgresConfig from a config.
+func (*PostgresConfigComponent) New(_ context.Context, c *PostgresConfig) (*PostgresConfig, error) {
```

Comment:
I am not _super_ familiar with the settings library, but I think this should return the thing that uses the configuration, in this case, the DB abstraction.

Reply:
That doesn't feel right.. We should discuss though

Reply:
Okay sounds good. I was making that call based off of https://github.com/asecurityteam/awsconfig-filterd/blob/master/pkg/filter/resource.go

and the components listed here: https://github.com/asecurityteam/transportd/tree/master/pkg/components

Reply:
Yeah I saw his Ben did that. It confused me for a bit... I think the component/config impls should be just that and no more.  In other words, no factories or domain struct creation in there, but you know more go than me. Learn me

Reply:
> no factories

But I think that is the intended purpose of settings.

```
// All configuration is driven by a struct that uses standard go types.
type Config struct {}
// The resulting element is virtually anything that you need it to be
// for the purposes of the system you are building.
type Result struct {}
// The component contract interfaces between the thing you want to make,
// the result, and the settings project. It is responsible for producing
// instances of the configuration struct with any default values populated.
// It is also used to construct new instances of the result using a
// populated configuration. No references to the settings project are
// required to create any part of this setup.
type Component struct {}
func (*Component) Settings() *Config { return &Config{} }
func (*Component) New(_ context.Context, c *Config) (*Result, error) {
    return &Result{}, nil
```

Reply:
Yeah, `settings` components are effectively factories, or factory factories, that get interface with a special configuration loader.